lib/engine-jit/src/artifact.rs

```diff
@@ -84,10 +84,15 @@ impl JITArtifact {
 
         let compiler = inner_jit.compiler()?;
 
+        let mut module = (*compile_info.module).clone();
+        let middlewares = compiler.get_middlewares();
+        middlewares.apply_on_module_info(&mut module);
```

Comment:
This `module` is a clone of the one in `translation`. That means that if the middleware introduces different memory or table styles, our maps `memory_styles` and `table_styles` will be out of date. I suppose the only way to do this today would be for the middleware to add a memory to a module that doesn't have one, or remove memory from one that does. Either way, maybe we should apply the middleware after `let translation = ` above, before we use `translation.module`?

Reply:
I'm just moving the logic, but your comment is completely on point (it was an issue also in previous code).
I'll try to move it up!

Reply:
Fixed it!